lib/Sabberworm/CSS/Value/Color.php

```diff
@@ -39,10 +39,31 @@ public static function parse(ParserState $oParserState) {
 			$sColorMode = $oParserState->parseIdentifier(true);
 			$oParserState->consumeWhiteSpace();
 			$oParserState->consume('(');
+
+			$bContainsVar = false;
 			$iLength = $oParserState->strlen($sColorMode);
 			for ($i = 0; $i < $iLength; ++$i) {
 				$oParserState->consumeWhiteSpace();
-				$aColor[$sColorMode[$i]] = Size::parse($oParserState, true);
+				if ($oParserState->comes('var')) {
+					$aColor[$sColorMode[$i]] = CSSFunction::parseIdentifierOrFunction($oParserState);
+					$bContainsVar = true;
+				} else {
+					$aColor[$sColorMode[$i]] = Size::parse($oParserState, true);
+				}
+
+				if ($bContainsVar && $oParserState->comes(')')) {
+					// With a var argument the function can have fewer arguments and the assignment to the color
+					// channels does not make sense
+					if ($i < $iLength - 1) {
+						// Add remaining characters to last item
+						$remainingChars = substr($sColorMode, $i + 1);
+						$aColor[$sColorMode[$i] . $remainingChars] = $aColor[$sColorMode[$i]];
+						unset($aColor[$sColorMode[$i]]);
```

Comment:
I do not really like this solution but, but it's necessary as the array keys are used in the render function below to define the color function. And because of the nature of `var()` it's not possible to easily tell which arguments are defined through the variable.

If it's not clear what I mean: `rgba(var(--rg), 255)` is valid CSS code and the variable value can contain a comma, so it's valid CSS and in the parser it's hard to tell whether it's for red and green. It's somehow possible to identify it, but this will result in quite some code to identify all combinations and I am not sure if its necessary.

Reply:
What if we parse colors that contain variables not as a `Color` instance but just an instance of a generic `CSSFunction`? I don’t see the additional value (pun intended) of using a specialized class when we can’t use it for any of the specializations we built it for…

Reply:
Good idea, I updated it as suggested.